### ISSUE TITLE
TypeScript prop typedef leads in declaration-colon-newline-after error

### ISSUE THREAD

**Issue by lumio** [styled-components/stylelint-processor-styled-components#158]
The following code as shown here:
```ts
import styled from 'styled-components';

import globalColors from '../../../utils/styles/colors';
import { SomeInterfaceType } from './types';

const GlobalFancySpan = styled.span`
    position: relative;
    display: inline-flex;
    align-items: center;

    width: 16.5rem;

    background: ${ ( props : SomeInterfaceType ) => (
        props.active ? globalColors.globalActiveBackground : globalColors.globalDefaultBackground
    ) };

    &:hover {
        background: ${ globalColors.globalActiveBackground };
    }

    transition: background-color .5s, padding .5s ease;

    & > div {
        visibility: hidden;
        opacity: 0;
        transition: visibility .1s .25s, opacity .25s;
    }

    &:hover > div {
        visibility: visible;
        opacity: 1;
        transition: visibility .1s, opacity .25s .1s;
    }
`;
```

leads to the following error
```
src/components/SomeComponent/styles.ts
 13:15  ✖  Expected newline after ":" with a    declaration-colon-newline-after
           multi-line declaration
```

I guess that the `props :` is interpreted as a new property.

---

Originally instead of `props.active...` I just posted `'#000'` which didn't reflect the real code snippet.
`globalColors` is an imported object that just holds color values.

**Comment by mxstbr** [comment]
The line and colon number (`13:15`) points to the `background:` bit, so I think our interpolation replacement might be going wrong there. Mind submitting this as a test to the codebase so we can take a closer look?

**Comment by lumio** [comment]
The line number is out of context. I left out some lines.
If I change it to the following below, the error is on the next line:

```ts
    background:
    ${ ( props : SomeInterfaceType ) => (
        '#000' /* some fancy stuff */
    ) };
```

**Comment by emilgoldsmith** [comment]
In that case I think you just need to use our interpolation tags as described in the docs. As it's on a new line I believe our "intelligent guessing" will treat it as a full declaration which is obviously wrong in your case. Try `/* sc-prop */`

**Comment by chinesedfan** [comment]
@lumio Can you provide the processor version and full stylelint config? I didn't find any problems with the latest 1.2.1 and following simple rules. Your first code sample is reasonable because it expects a newline after `background:`. And the second code returns a warning `'Expected indentation of 8 spaces (indentation)'`, which is also correct. Please remind me if I missed something.

```
const rules = {
  "declaration-colon-newline-after": "always-multi-line",
  indentation: 4
}
```

@emilgoldsmith Remember what we talked in [#130](https://github.com/styled-components/stylelint-processor-styled-components/issues/130#issuecomment-349226323) that interpolation tags do not work in TS.

**Comment by emilgoldsmith** [comment]
> @emilgoldsmith Remember what we talked in #130 that interpolation tags do not work in TS.

Ahh yes, if you have the newest version of the `typescript-eslint-parser` it won't include the comments, right? Which in that case breaks our interpolation tags.
But you can't reduce OP's problem at the moment right?

**Comment by chinesedfan** [comment]
> If I change it to the following below, the error is on the next line

@emilgoldsmith Yes, I don't understand what `the error` means. If it means the same error, I failed to reproduce. So hope more information can be provided.

**Comment by lumio** [comment]
Hey guys! I'm super sorry for the confusion. I tested that very specific code snippet - and yes, it works without any problem.

I updated my initial post with the real problem. I somehow thought that the error was independent of the function body.

**Comment by emilgoldsmith** [comment]
Thanks @lumio 

@chinesedfan you up for looking at it again with the new information?

**Comment by chinesedfan** [comment]
@lumio Same results, even if I deleted the additional right parenthesis in your updated snippet. Excuse me, what's your expected results? I think the warning of `declaration-colon-newline-after` is right.

**Comment by lumio** [comment]
Hi @chinesedfan. Thanks for looking into it.
So my expected result was, that stylelint interprets the following lines below as `background: #hexcolor;`:

```
    background: ${ ( props : SomeInterfaceType ) => (
        props.active ? globalColors.globalActiveBackground : globalColors.globalDefaultBackground
    ) };
```

Digging into it a little deeper I figured out the problem. And you are right, `declaration-colon-newline-after` works as expected.
I just needed to indent the lines properly. So my fault after all
